pkgs/applications/networking/instant-messengers/ripcord/default.nix

```
@@ -0,0 +1,69 @@
+{ lib, mkDerivation, fetchurl, makeFontsConf, appimageTools,
+  qtbase, qtsvg, qtmultimedia, qtwebsockets, qtimageformats,
+  autoPatchelfHook, desktop-file-utils, imagemagick, makeWrapper,
+  twemoji-color-font, xorg, libsodium, libopus, libGL, zlib, alsaLib }:
+
+mkDerivation rec {
+  pname = "ripcord";
+  version = "0.4.23";
+
+  src = let
+    appimage = fetchurl {
+      url = "https://cancel.fm/dl/Ripcord-${version}-x86_64.AppImage";
+      sha256 = "0395w0pwr1cz8ichcbyrsscmm2p7srgjk4vkqvqgwyx41prm0x2h";
+      name = "${pname}-${version}.AppImage";
+    };
+  in appimageTools.extractType2 {
```

```suggestion
  in appimageTools.extract {
```

looks like they are aliases of each other, and the imagetype is extracted from the image metadata:
```
  extractType1 = extract;
  extractType2 = extract;
```
```
    buildCommand = ''
      # https://github.com/AppImage/libappimage/blob/ca8d4b53bed5cbc0f3d0398e30806e0d3adeaaab/src/libappimage/utils/MagicBytesChecker.cpp#L45-L63
      eval $(r2 $src -nn -Nqc "p8j 3 @ 8" |
        jq -r '{appimageSignature: (.[:-1]|implode), appimageType: .[-1]}|
          @sh "appimageSignature=\(.appimageSignature) appimageType=\(.appimageType)"')
```

---

true :D